src/dmd/dclass.d

```diff
@@ -683,6 +683,14 @@ extern (C++) class ClassDeclaration : AggregateDeclaration
         checkOverlappedFields();
     }
 
+    /**************
+     * Returns: true if there's a __monitor field
+     */
+    final bool hasMonitor()
+    {
+        return classKind == ClassKind.d || classKind == ClassKind.objc;
```

Comment:
Objective-C classes should not have a monitor field.

Reply:
Hmm, but I guess it does.

Reply:
ok

Reply:
I guess it safest to leave it like it is.